frontend/src/presentation/components/QuestionFilters.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { useState, useMemo, useCallback } from 'react';
+import { Select, Input, Tag, Dropdown, Button } from 'antd';
+import { DownOutlined } from '@ant-design/icons';
+import styles from './QuestionFilters.module.css';
+
+const { Option } = Select;
+const { CheckableTag } = Tag;
+
+interface QuestionFiltersProps {
+  allCategories: string[];
+  onFiltersChange: (filters: {
+    selectedDifficulty: string;
+    selectedCategories: string[];
+    searchTerm: string;
+  }) => void;
+}
+
+export const QuestionFilters: React.FC<QuestionFiltersProps> = ({
+  allCategories,
+  onFiltersChange,
+}) => {
+  const [selectedDifficulty, setSelectedDifficulty] = useState<string>('All');
+  const [selectedCategories, setSelectedCategories] = useState<string[]>([]);
+  const [categorySearchTerm, setCategorySearchTerm] = useState('');
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const triggerFiltersChange = useCallback(
+    (filters: {
+      selectedDifficulty: string;
+      selectedCategories: string[];
+      searchTerm: string;
+    }) => {
+      if (onFiltersChange) {
+        onFiltersChange(filters);
```

Comment:
If you're checking whether or not `onFiltersChange` exists, I think you can just do `onFiltersChange?.()`

For stricter checks, we can do `typeof(onFiltersChange) == "function"`. It's unlikely that we'll pass in anything else other than a function for `onFiltersChange` so I think yours is fine if you dont wanna change

Reply:
That's cleaner, I'll use that instead. 

No need to change, the type is already ensured by QuestionFiltersProps to be a specific function with specific parameters. The only things that can pass the 'check' is null and actual functions.